app/middlewares/validators/users.js

```diff
@@ -0,0 +1,28 @@
+const UserModel = require('../../models/user');
+const errors = require('../../errors');
+
+exports.checkExistingEmail = async email => {
+  const user = await UserModel.findOne({ where: { email } });
+
+  if (user) throw errors.badRequestError('User already exists');
+};
+
+exports.checkPasswordRestriction = password => {
+  const isAlphanumericRegex = /[^a-z\d]/i;
+
+  const isAlphanumeric = isAlphanumericRegex.test(password);
+
+  if (!isAlphanumeric || password.length >= 8) {
+    throw errors.badRequestError('Password should have 8 characters minimum');
+  }
+};
+
+exports.checkEmailRestriction = email => {
+  const isAWoloxDomainRegex = /^[a-zA-Z]+.[a-zA-Z]+@wolox.+((co)|(ar)|(mx))$/;
+
+  const isAWoloxDomain = isAWoloxDomainRegex.test(email);
+
+  if (!isAWoloxDomain) {
+    throw errors.badRequestError('The domain is not part of Wolox');
+  }
+};
```

Comment:
Although both these functions are totally fine, you can do all this validations using library made for this, check schema validation on the `express-validator` package or another interesting one is `joi `

Reply:
I was gonna do it that way but i saw a card called Middleware where this is a task to do there, so i'm trying manual and then improving the code by using the library when i get to that card.